packages/datadog-plugin-net/src/index.js

```diff
@@ -8,9 +8,15 @@ function createWrapConnect (tracer, config) {
     return function connectWithTrace () {
       const scope = tracer.scope()
       const options = getOptions(arguments)
+      const lastIndex = arguments.length - 1
+      const callback = arguments[lastIndex]
 
       if (!options) return connect.apply(this, arguments)
 
+      if (typeof callback === 'function') {
```

Comment:
In order to ensure this bind happens property, shouldn't it come before the `if (!options)` early return?

Reply:
The callback needs to be bound to the parent because otherwise it runs in the original `connect` that is bound to its span. If we never bind `connect`, then it's not needed to bind the callback since it will automatically run in the context of the parent.